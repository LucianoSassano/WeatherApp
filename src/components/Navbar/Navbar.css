/* LARGE SWITCH */

.custom-switch.custom-switch-lg {
    padding-bottom: 1rem;
    padding-left: 2.25rem;
}
.custom-switch.custom-switch-lg .custom-control-label {
    padding-left: 0.75rem;
    padding-top: 0.15rem;
}
.custom-switch.custom-switch-lg .custom-control-label::before {
    border-radius: 1rem;
    height: 1.5rem;
    width: 2.5rem;
}
.custom-switch.custom-switch-lg .custom-control-label::after {
    border-radius: 0.65rem;
    height: calc(1.5rem - 4px);
    width: calc(1.5rem - 4px);
}
.custom-switch.custom-switch-lg .custom-control-input:checked ~ .custom-control-label::after {
    transform: translateX(1rem);
}








@media all and (min-width: 280px)
{
   .navbar-container
   {
       margin-bottom: 28vh;
       font-size: 6vw;
   }

  
}


@media all and (min-width: 320px)
{
   .navbar-container
   {
       margin-bottom: 20vh;
       font-size: 6vw;
   }

   .single-search-container input{

    height: 4vh;
    width: 90vw;
    margin-right: auto;
    margin-left: auto;
    font-size: 4.0vw;

    }

   .single-search-container button{
       
    width: 22vw;
    margin-left: 1vw;
    font-size: 4.0vw;

    }


    .dual-search-form
    {
        display: flex;
        flex-wrap: wrap;
        
        
    }

    .dual-search-form input{
        
        width: 90vw;
        height: 4.0vh; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 4.0vw;
        display: flex;
       
        margin-bottom: 0.5vh;
        

    }

    .dual-search-form button{

        font-size: 4.0vw;
        height: 4vh;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;
        
        
    }


    
}


@media all and (min-width: 375px)
{
   .navbar-container
   {
       margin-bottom: 28vh;
       font-size: 6vw;
   }

   .single-search-container input{

    height: 4vh;
    width: 90vw;
    margin-right: auto;
    margin-left: auto;
    font-size: 4.0vw;

    }

   .single-search-container button{
    width: 22vw;
    margin-left: 1vw;
    font-size: 4.0vw;


    }


    .dual-search-form
    {
        display: flex;
        flex-wrap: wrap;
        
        
    }

    .dual-search-form input{
        
        width: 90vw;
        height: 4.0vh; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 4.0vw;
        display: flex;
       
        margin-bottom: 0.5vh;
        

    }

    .dual-search-form button{

        font-size: 4.0vw;
        height: 4vh;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;
        
        
    }

}


@media all and (min-width: 425px)
{
   .navbar-container
   {
       margin-bottom: 35vh;
       font-size: 6vw;
       
    }


   .single-search-container input{

    height: 5vh;
    width: 90vw;
    margin-right: auto;
    margin-left: auto;
    font-size: 4.0vw;

    }

   .single-search-container button{
    
    height: 5vh;
    width: 22vw;
    margin-left: 1vw;
    font-size: 4.0vw;

    }


    .dual-search-form
    {
        display: flex;
        flex-wrap: wrap;
        
        
    }

    .dual-search-form input{
        
        width: 90vw;
        height: 9.0vh; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 4.0vw;
        display: flex;
       
        margin-bottom: 0.5vh;
        

    }

    .dual-search-form button{

        font-size: 4.0vw;
        height: 5vh;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;
        
        
    }

   
}


@media all and (min-width: 768px)
{

    .navbar-container
    {
        
        font-size: 3vh;
        margin-bottom: 22vh;
    }

    .navbar-container label{
        font-size: 2.5vw;
        
    }

    

    .navbar
    {
        height: 17vh;
    }


    .single-search-container input{

        height: 4.0vh; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 2.2vw;
        

    }

    .single-search-container button{

        font-size: 2.2vw;
        height: 4vh;
        width: 10vw;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;


    }


    .dual-search-form
    {
        display: flex;
        flex-wrap: wrap;
        
        
    }

    .dual-search-form input{
        
        height: 4.0vh; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 2.2vw;
        
    }

    .dual-search-form button{

        font-size: 2.2vw;
        height: 4vh;
        width: 10vw;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;
        
        
    }

    
}


@media all and (min-width: 1024px)
{

    .navbar-container
    {
        
        font-size: 3vh;
        margin-bottom: 12vh;
    }
    
    .navbar-container label{
        font-size: 2.0vw;
        
    }

    .navbar
    {
        height: 17vh;
       
    }

    .single-search-container input{

        height: 4.0vh; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 1.5vw;

    }

    .single-search-container button{

        font-size: 1.5vw;
        height: 4vh;
        width: 10vw;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;


    }


    .dual-search-form
    {
        display: flex;
        flex-wrap: wrap;
        
        
    }

    .dual-search-form input{
        
        height: 4.0vh; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 1.5vw;

    }

    .dual-search-form button{

        font-size: 1.5vw;
        height: 4vh;
        width: 10vw;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;
        
        
    }

    
}

@media all and (min-width: 1440px)
{

    .navbar-container
    {
        
        font-size: 3vh;
        margin-bottom: 5vh;
    }

    .navbar-container label{
        font-size: 1.5vw;
        
    }

    .navbar
    {
        
        height: 16vh;
    }


    .single-search-container input{

        height: 4.4vh; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 1.1vw;

    }

    .single-search-container button{

        font-size: 1.1vw;
        height: 4.4vh;
        width: 10vw;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;


    }

     
    .dual-search-form
    {
        display: flex;
        flex-wrap: wrap;
        flex-direction: row;
        justify-content: flex-end;
        margin-left: 3vw;
        
        
    }


    .dual-search-form input{
        
        height: 4.4vh;
        font-size: 1.1vw; 
        margin-right: 0.5vw;
        margin-left: 0.5vw;

    }

    .dual-search-form button{

        height: 4.4vh;
        font-size: 1.1vw;
        width: 10vw;
        margin-left: 1vw;
        display: flex;
        justify-content: center;
        align-items: center;
        
        
        
    }

    
}


@media all and (min-width: 1920px)
{

    .navbar
    {
        height: 16vh;
        margin-bottom: 5vh;

    }

    .navbar-container
    {
    
        display: flex;
        align-items: center;    
    
    }


    .navbar-container label{
        font-size: 1.5vw;
        
    }


    .custom-switch.custom-switch-xl {

        padding-bottom: 1rem;
        padding-left: 2.25rem;
        
    
    }


    .custom-switch.custom-switch-xl .custom-control-label {

        padding-left: 2rem;
        padding-top: 0.5rem;
        
        
    }
    .custom-switch.custom-switch-xl .custom-control-label::before {

        border-radius: 1rem;
        height: 2rem;
        width: 3.5rem;
    }

    .custom-switch.custom-switch-xl .custom-control-label::after {

        border-radius: 2rem;
        height: calc(2rem - 4px);
        width: calc(2rem - 4px);
    }

    .custom-switch.custom-switch-xl .custom-control-input:checked ~ .custom-control-label::after {

        transform: translateX(1.5rem);
    
    }


    .single-search-container input{

        height: 4vh;
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 2.0vh;

    }

    .single-search-container button{

        height: 4vh;
        font-size: 2vh;
        width: 15vw;
        margin-left: 1vw;


    }

    
    .dual-search-form
    {
        display: flex;
        flex-wrap: wrap;
        flex-direction: row;
        justify-content: flex-end;
        margin-left: 3vw;
        
        
    }


    .dual-search-form input{
        
        height: 4vh;
        margin-right: 0.5vw;
        margin-left: 0.5vw;
        font-size: 2.0vh;

    }

    .dual-search-form button{

        font-size: 2vh;
        width: 15vw;
        margin-left: 1vw;
        height: 4vh;
        
        
    }

    
}