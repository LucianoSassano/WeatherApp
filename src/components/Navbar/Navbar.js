import React, { useState } from "react";
import "../Navbar/Navbar.css";

const Navbar = (props) => {
  const [searchMode, setSerchMode] = useState(false);

  const handleSerchMode = () => {
    let mode = !searchMode;
    setSerchMode(mode);
  };

  return (
    <>
      <nav className="navbar fixed-top navbar-expand-lg navbar-light bg-light">
        <div  id="navbarSupportedContent">
          <ul className="navbar-nav mr-auto">
            <li className="nav-item ">
              <a className="nav-link" href="./">
                Home
              </a>
            </li>
            <a className="nav-link " href="./favorites">
              Favorites
            </a>
          </ul>
          <div className="searchForm">
            <div className="custom-control custom-switch">
              <input
                onClick={handleSerchMode}
                type="checkbox"
                className="custom-control-input"
                id="customSwitch1"
              />
              <label className="custom-control-label" htmlFor="customSwitch1">
                Dual Location Serch
              </label>
            </div>
            {searchMode ? (
              <form className="form-inline my-2 my-lg-0">
                <input
                  onChange={props.onChangeSingle}
                  className="form-control mr-sm-2"
                  type="search"
                  placeholder="Search a location"
                  aria-label="Search a location"
                  onKeyPress={props.handleDualEnter}
                />
                <input
                  onChange={props.onChangeDual}
                  className="form-control mr-sm-2"
                  type="search"
                  placeholder="Search a location"
                  aria-label="Search a location"
                  onKeyPress={props.handleDualEnter}
                />
                <button
                  className="btn btn-outline-success my-2 my-sm-0"
                  type="button"
                  onClick={props.handleDualSubmit}
                >
                  Search
                </button>
              </form>
            ) : (
              <form className="form-inline my-2 my-lg-0">
                <input
                  onChange={props.onChangeSingle}
                  onKeyPress={props.handleEnter}
                  className="form-control mr-sm-2"
                  type="search"
                  placeholder="Search a location"
                  aria-label="Search a location"
                />
                <button
                  className="btn btn-outline-success my-2 my-sm-0"
                  type="button"
                  onClick={props.handleSearchSubmit}
                >
                  Search
                </button>
              </form>
            )}
          </div>
        </div>
      </nav>
    </>
  );
};

export default Navbar;
