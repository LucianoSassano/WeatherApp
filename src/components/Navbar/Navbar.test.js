import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';


import Navbar from './Navbar.js';

afterEach(cleanup);

it('Navbar renders correctly', () => {

    const div = document.createElement("div");
    ReactDOM.render(<Navbar></Navbar>, div)
})


it('Home button renders correctly', () => {

    const { getByTestId } = render(<Navbar></Navbar>)
    expect(getByTestId('navbarTestId')).toHaveTextContent('Home')
})

it('Favorites button renders correctly', () => {

    const { getByTestId } = render(<Navbar></Navbar>)
    expect(getByTestId('navbarTestId')).toHaveTextContent('Favorites')
})


it('Dual Location Search button renders correctly', () => {

    const { getByTestId } = render(<Navbar></Navbar>)
    expect(getByTestId('navbarTestId')).toHaveTextContent('Dual Location Search')
})

it('Dual Location Search button renders correctly', () => {

    const { getByTestId } = render(<Navbar></Navbar>)
    expect(getByTestId('navbarTestId')).toHaveTextContent('Search')
})

it('Dual Location Search Toggle Button click', () => {

    render(

        <div className="custom-control custom-switch">
            <input

                data-testid="dualSearchBtn"
                type="checkbox"
                className="custom-control-input"
                id="customSwitch1"
                aria-label="dual location search toggle button"
            />
            <label className="custom-control-label" htmlFor="customSwitch1">
                Dual Location Search
              </label>
        </div>

    )

    userEvent.click(screen.getByTestId('dualSearchBtn'))
    expect(screen.getByTestId('dualSearchBtn')).toBeChecked()

})



it("Navbar component matches snapshot", () => {

    const navbar = renderer.create(<Navbar></Navbar>).toJSON();
    expect(navbar).toMatchSnapshot();

})


