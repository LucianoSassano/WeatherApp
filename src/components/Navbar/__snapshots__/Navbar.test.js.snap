// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `
<nav
  aria-label="navbar"
  className="navbar fixed-top navbar-expand-lg navbar-light bg-light"
  data-testid="navbarTestId"
>
  <div
    className="container-fluid"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        aria-label="Home button"
        className="nav-item "
      >
        <a
          className="nav-link"
          href="./"
        >
          Home
        </a>
      </li>
      <li
        aria-label="Favorites button"
        className="nav-item"
      >
        <a
          className="nav-link "
          href="./favorites"
        >
          Favorites
        </a>
      </li>
    </ul>
    <div
      className="searchForm"
    >
      <div
        className="custom-control custom-switch"
      >
        <input
          aria-label="dual location search toggle button"
          className="custom-control-input"
          id="customSwitch1"
          onClick={[Function]}
          type="checkbox"
        />
        <label
          className="custom-control-label"
          data-testid="dualSearchBtn"
          htmlFor="customSwitch1"
        >
          Dual Location Search
        </label>
      </div>
      <form
        className="form-inline my-2 my-lg-0"
      >
        <input
          aria-label="Search a location"
          className="form-control mr-sm-2"
          placeholder="Search a location"
          type="search"
        />
        <button
          aria-label="search"
          className="btn btn-outline-success my-2 my-sm-0"
          type="button"
        >
          Search
        </button>
      </form>
    </div>
  </div>
</nav>
`;
